ihmc-perception/src/main/java/us/ihmc/perception/gpuHeightMap/RapidHeightMapExtractorCUDA.java

```diff
@@ -0,0 +1,529 @@
+package us.ihmc.perception.gpuHeightMap;
+
+import org.bytedeco.cuda.cudart.CUstream_st;
+import org.bytedeco.cuda.cudart.dim3;
+import org.bytedeco.cuda.global.cudart;
+import org.bytedeco.javacpp.FloatPointer;
+import org.bytedeco.javacpp.Pointer;
+import org.bytedeco.opencv.global.opencv_core;
+import org.bytedeco.opencv.opencv_core.GpuMat;
+import org.bytedeco.opencv.opencv_core.Mat;
+import org.bytedeco.opencv.opencv_core.Rect;
+import org.bytedeco.opencv.opencv_core.Scalar;
+import us.ihmc.euclid.referenceFrame.ReferenceFrame;
+import us.ihmc.euclid.transform.RigidBodyTransform;
+import us.ihmc.euclid.tuple3D.Point3D;
+import us.ihmc.euclid.tuple3D.interfaces.Tuple3DReadOnly;
+import us.ihmc.perception.camera.CameraIntrinsics;
+import us.ihmc.perception.cuda.CUDAKernel;
+import us.ihmc.perception.cuda.CUDAProgram;
+import us.ihmc.perception.cuda.CUDATools;
+import us.ihmc.perception.heightMap.TerrainMapData;
+import us.ihmc.perception.neural.HeightMapAutoencoder;
+import us.ihmc.perception.tools.PerceptionMessageTools;
+import us.ihmc.robotics.robotSide.RobotSide;
+import us.ihmc.robotics.robotSide.SideDependentList;
+import us.ihmc.sensorProcessing.heightMap.HeightMapData;
+import us.ihmc.sensorProcessing.heightMap.HeightMapParameters;
+import us.ihmc.sensorProcessing.heightMap.HeightMapTools;
+
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import static org.bytedeco.cuda.global.cudart.cudaStreamSynchronize;
+
+public class RapidHeightMapExtractorCUDA
+{
+   private GpuMat inputDepthImage;
+   private GpuMat transformedInputDepthImage;
+   private GpuMat localHeightMapImage;
+   private GpuMat globalHeightMapImage;
+   private GpuMat globalHeightVarianceImage;
+   private GpuMat terrainCostImage;
+   private GpuMat contactMapImage;
+   private GpuMat sensorCroppedHeightMapImage;
+   private GpuMat sensorCroppedTerrainCostImage;
+   private GpuMat sensorCroppedContactMapImage;
+
+   private boolean initialized = false;
+   private int mode = 1; // 0 -> Ouster, 1 -> Realsense
```

Comment:
Doesn't 1 mean pinhole in general? Like it should work with any generic depth sensor, right?

Reply:
The way it's calculated back_project_perspective/back_project_spherical is different for ouster and different for the realsense, how we find the center and calculate yaw and pitch index

Reply:
I meant that 1 isn't specific for a realsense. You can use it with any pinhole/fisheye style sensor